pkg/model/context.go

```diff
@@ -62,19 +63,69 @@ func (s *Context) InspectEtcdExistingState(c *Config) (api.EtcdExistingState, er
 	if err != nil {
 		return state, fmt.Errorf("failed to check for existence of etcd cloud-formation stack: %v", err)
 	}
-	// when the Etcd stack does not exist but we have existing etcd instances then we need to enable the
-	// etcd migration units.
-	if !state.StackExists {
-		if state.EtcdMigrationExistingEndpoints, err = s.lookupExistingEtcdEndpoints(c); err != nil {
-			return state, fmt.Errorf("failed to lookup existing etcd endpoints: %v", err)
+	// when the Etcd stack exists we need to check for the MajorMinor version of Etcd running and trigger a migration if different to ours.
+	if state.StackExists {
+		if state.EtcdMigrationEnabled, err = s.isAMajorEtcdUpgrade(c); err != nil {
+			return state, fmt.Errorf("failed to check existing etcd major minor version: %v", err)
 		}
-		if state.EtcdMigrationExistingEndpoints != "" {
-			state.EtcdMigrationEnabled = true
+		if state.EtcdMigrationEnabled {
+			logger.Warn("Performing a Major Etcd Version Upgrade: -")
+			logger.Warn("To do this we will spin up new etcd servers and then export the existing kubernetes state to them.")
+			logger.Warn("There will be cluster disruption until all of your existing controllers have rolled.")
+			logger.Warn("If the cloudformation update fails (at any point) then we will roll back to the original etcd servers.")
+			logger.Warn("You MAY lose/rollback changes that are made to the cluster AFTER the etcd export has been performed!")
+			logger.Warn("This operation is best scheduled for a quiet time or in an outage window.")
+			if state.EtcdMigrationExistingEndpoints, err = s.lookupExistingEtcdEndpoints(c); err != nil {
```

Comment:
If we're going to bomb out on `lookupExistingEtcdEndpoints`, should we try and do that before telling the user that we're `Performing a Major Etcd Upgrade`? 

As a layman, I might be concerned that something may have changed before the thing blew up.

Reply:
Good idea - I've swapped it around so we announce the migration only if the lookup is successful.